plugins/army-controller-sanity.cpp

```diff
@@ -0,0 +1,133 @@
+#include "Debug.h"
+#include "PluginManager.h"
+
+#include "df/world.h"
+#include "df/army_controller.h"
+#include "df/army.h"
+#include "df/historical_entity.h"
+#include "df/unit.h"
+#include "df/global_objects.h"
+
+#include <unordered_set>
+
+DFHACK_PLUGIN("army-controller-sanity");
+DFHACK_PLUGIN_IS_ENABLED(is_enabled);
+
+REQUIRE_GLOBAL(army_controller_next_id);
+REQUIRE_GLOBAL(world);
+REQUIRE_GLOBAL(pause_state);
+
+namespace DFHack {
+    DBG_DECLARE(army_controller_sanity, log, DebugCategory::LWARNING);
+}
+
+namespace {
+    bool checkArmyControllerSanity()
+    {
+        // army controllers are found:
+        //   in viewscreen_worldst (army_controller (list), last_hover_ac) (not checked)
+        //   in entitst (army_controller (list))
+        //   in armyst (controller)
+        //   in unitst (army_controller)
+        //
+        // master list is in army_controller_handlerst
+
+        static size_t last_ac_vec_size = 0;
+        static int last_army_controller_next_id = 0;
+
+        if (last_army_controller_next_id == *df::global::army_controller_next_id &&
```

Comment:
explicit `df::global` usage in plugins is a warning sign that REQUIRE_GLOBAL was forgotten. this plugin adds REQUIRE_GLOBAL correctly, so the df::global prefix should be elided